pkg/tlsutil/tls.go

```diff
@@ -188,16 +196,54 @@ func (scg *SDKCertGenerator) GenerateCert(cr runtime.Object, service *v1.Service
 		}
 		return appSecret, caConfigMap, caSecret, nil
 	} else {
+		var caKeyData, caCertData []byte
+		var err error
+		var caKey *rsa.PrivateKey
+		var caCert *x509.Certificate
 		// case: both CA and Application TLS assets don't exist.
-		caKey, err := newPrivateKey()
-		if err != nil {
-			return nil, nil, nil, err
+		if config.CAKey != "" && config.CACert != "" {
+			// custom CA provided by the user.
+			caKeyData, err = ioutil.ReadFile(config.CAKey)
+			if err != nil {
+				return nil, nil, nil, ErrReadingCAKey
+			}
+
+			caCertData, err = ioutil.ReadFile(config.CACert)
+			if err != nil {
+				return nil, nil, nil, ErrReadingCACert
+			}
+
+			caKey, err = parsePEMEncodedPrivateKey(caKeyData)
+			if err != nil {
+				return nil, nil, nil, ErrParsingCAKey
+			}
+
+			caCert, err = parsePEMEncodedCert(caCertData)
+			if err != nil {
+				return nil, nil, nil, ErrParsingCACert
+			}
+
 		}
-		caCert, err := newSelfSignedCACertificate(caKey)
-		if err != nil {
-			return nil, nil, nil, err
+
+		if config.CAKey != "" || config.CACert != "" {
+			// if only one of the custom CA Key or Cert is provided
+			return nil, nil, nil, ErrCAKeyAndCACertReq
+		} else {
+			// If no custom CAKey and CACert are provided we have to generate them
+			caKey, err := newPrivateKey()
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			caCert, err := newSelfSignedCACertificate(caKey)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+
+			caKeyData = encodePrivateKeyPEM(caKey)
+			caCertData = encodeCertificatePEM(caCert)
 		}
-		caSecret, caConfigMap := toCASecretAndConfigmap(caKey, caCert, caSecretAndConfigMapName)
+
+		caSecret, caConfigMap := toCASecretAndConfigmap(caKeyData, caCertData, caSecretAndConfigMapName)
```

Comment:
This code would create the config map/secret for the custom CA. I thought we decided we should not do this?